widget/tree_internal_test.go

```diff
@@ -278,6 +278,35 @@ func TestTree_MinSize(t *testing.T) {
 	}
 }
 
+func TestTree_Selection(t *testing.T) {
+	data := make(map[string][]string)
+	widget.AddTreePath(data, "A", "B")
+	tree := NewTreeWithStrings(data)
+
+	tree.Refresh() // Force layout
+
+	selection := make(chan string, 2)
+	tree.OnSelectionChanged = func(uid string) {
+		selection <- uid
+	}
+
+	tree.SetSelection("A")
+	select {
+	case s := <-selection:
+		assert.Equal(t, "A", s)
+	case <-time.After(1 * time.Second):
+		assert.Fail(t, "Selection should have occured")
+	}
+
```

Comment:
Can you check that the Selection() method returns the correct result after the selection is changed as well?

Reply:
Done!